File: app/src/main/java/com/example/networkaplication/persistance/repository/database/dao/MovieQueryDao.java
package com.example.networkaplication.persistance.repository.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Transaction;

import com.example.networkaplication.persistance.model.MovieQuery;

import java.util.List;

@Dao
public abstract class MovieQueryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)

    public abstract void insert (MovieQuery movieQuery);

    @Transaction
    @Query("SELECT * FROM movie_query WHERE name LIKE :search")
    public abstract List<MovieQuery> findAllLike (String search);

    @Query("SELECT * FROM movie_query WHERE name=:name")
    public abstract MovieQuery findByTitle (String name);

    @Delete
    public abstract void delete(MovieQuery movieQuery);

    @Query("DELETE FROM movie_query")
    public abstract void deleteAll();
}

File: app/src/main/java/com/example/networkaplication/persistance/repository/database/dao/MovieQueryDao.kt
package com.example.networkaplication.persistance.repository.database.dao


import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Transaction

import com.example.networkaplication.persistance.model.MovieQuery

@Dao
abstract class MovieQueryDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insert(movieQuery: MovieQuery)

    @Transaction
    @Query("SELECT * FROM movie_query WHERE name LIKE :search")
    abstract fun findAllLike(search: String): List<MovieQuery>

    @Query("SELECT * FROM movie_query WHERE name=:name")
    abstract fun findByTitle(name: String): MovieQuery

    @Delete
    abstract fun delete(movieQuery: MovieQuery)

    @Query("DELETE FROM movie_query")
    abstract fun deleteAll()
}
